cluster-autoscaler/core/utils/pod_schedulable.go

```diff
@@ -73,11 +77,19 @@ func (podMap PodSchedulableMap) Get(pod *apiv1.Pod) (*simulator.PredicateError,
 // Set sets scheduling info for given pod in PodSchedulableMap
 func (podMap PodSchedulableMap) Set(pod *apiv1.Pod, err *simulator.PredicateError) {
 	ref := drain.ControllerRef(pod)
-	if ref == nil {
+	if ref == nil || pod_utils.IsDaemonSetPod(pod) {
 		return
 	}
 	uid := string(ref.UID)
-	podMap[uid] = append(podMap[uid], PodSchedulableInfo{
+	pm := podMap[uid]
+	if len(pm) >= maxPodsPerOwnerRef {
```

Comment:
The end result is roughly the same<sup>1</sup>, but it's implemented in a completely different way (limiting list size vs short-circuiting lookups). Wouldn't it be better to use a similar implementation? Arguably it makes code more readable (more clear the logic is the same) and (admittedly, minimally) reduces the risk of diverging implementations.

<sup>1</sup> technically with this version we still do 10*#pods comparisons and with the other we don't, but I'm not sure this makes practical difference).

Reply:
+1 to not worth the time :)